python/lsst/verify/compatibility/metricRegistry.py

```diff
@@ -0,0 +1,206 @@
+    `~lsst.pex.config.Config` class with one field, ``configs``. ``configs``
+    is a `~lsst.pex.config.ConfigDictField` that may have any number of
+    configs attached to it. The field will create multiple
+    `~lsst.verify.compatibility.MetricTask` objects, one for each config
+    provided. See :lsst-task:`~lsst.verify.compatibility.MetricsControllerTask`
+    for an example how to use ``configs``.
+
+    Examples
+    --------
+    The decorator is applied at the class definition:
+
+    >>> from lsst.verify.compatibility import registerMultiple, MetricTask
+    >>> @registerMultiple("reusable")
+    ... class ReusableMetricTask(MetricTask):
+    ...     pass
+    """
+    def wrapper(taskClass):
+        if issubclass(taskClass, MetricTask):
+            # TODO: if MetricRegistry is phased out, simply return taskClass
+            # instead of removing the decorator
+            MetricRegistry.registry.register(
+                name,
+                _MultiConfigFactory(taskClass),
+                ConfigClass=_makeMultiConfig(taskClass.ConfigClass))
+            return taskClass
+        else:
+            raise ValueError("%r is not a %r" % (taskClass, MetricTask))
+    return wrapper
+
+
+def _makeMultiConfig(configClass):
+    """A factory function for creating a config for registerMultiple.
+
+    Parameters
+    ----------
+    configClass : `lsst.verify.compatibility.MetricTask.ConfigClass`-type
+        The type of task config to be stored inside the new config. Subclasses
+        of ``configClass`` will **NOT** be supported (this is a limitation of
+        `~lsst.pex.config.ConfigDictField`).
+
+    Returns
+    -------
+    multiConfig : `lsst.pex.config.Config`-type
+        A `~lsst.pex.config.Config` class containing the following fields:
+
+        configs : `lsst.pex.config.ConfigDictField`
+            A field that maps `str` to ``configClass``. The keys are arbitrary
+            and can be chosen for user convenience.
+    """
+    class MultiConfig(Config):
+        configs = ConfigDictField(
+            keytype=str, itemtype=configClass, optional=False,
+            default={},
+            doc="A collection of multiple configs to create multiple items "
+                "of the same type.")
+
+    return MultiConfig
+
+
+class _MultiConfigFactory:
+    """A factory class for creating multiple `MetricTask` objects at once.
+
+    Parameters
+    ----------
+    configurableClass : `lsst.verify.compatibility.MetricTask`-type
+        The type of configurable created by `__call__`.
+    """
+    def __init__(self, configurableClass):
+        self._configurableClass = configurableClass
+
+    def __call__(self, config, **kwargs):
+        """Create the configured task(s).
+
+        Parameters
+        ----------
+        config : ``_makeMultiConfig(configurableClass.ConfigClass)``
+            A config containing multiple configs for ``configurableClass`.
```

Comment:
The definition would need to be more specific than that, because the `ConfigDictField` needs to be configured in a specific way. This admittedly non-standard notation was the easiest way to avoid repeating myself (though I could add an "output of" to be more explicit).

As for configs -> `configs`, that seems kind of confusing to me. I meant "config" in the general sense.